src/navigation/navigationhandler_lua.inl

```diff
@@ -254,43 +285,96 @@ namespace {
     );
 }
 
+// @TODO: Should this be a concrete Type in the documentation? Could also be used in bind
+// joystick functions above//
+struct [[codegen::Dictionary(JoystickAxis)]] JoystickAxis {
+    // The current type of axis binding
+    std::string type;
+
+    // Whether the values are inverted
+    bool invert;
+
+    // The type of joystick that this axis represents on the controller - either
+    // `\"JoystickLike\"` or `\"TriggerLike\"`. A joystick is `\"TriggerLike\"` if it
+    // can only be pressed or pushed in one direction. A `\"JoystickLike\"` axis can
+    // be pushed in two directions; for example, left and right, or up and down.
+    std::string joystickType;
+
+    // Whether or not this axis is “sticky”. An axis is “sticky” if, when you let go of
+    // it, the values it represents in the software do not go back to the default.
+    bool isSticky;
+
+    // Whether the camera movement for the axis is reversed. In the case of a
+    // `\"JoystickLike\"` axis, this is the same as inverting the axis. However, in the
+    // case of a `\"TriggerLike\" axis, this can reverse the camera movement for the
+    // trigger.
+    bool flip;
+
+    // Sensitivity value for this axis
+    double sensitivity;
+
+    // The identifier (URI) of the property that is bound to this axes, if one is
+    std::optional<std::string> propertyUri;
+
+    // If a property is bound to this axis, this is the min value that can be set using
+    // the joystick input
+    std::optional<float> minValue;
+
+    // If a property is bound to this axis, this is the max value that can be set using
+    // the joystick input
+    std::optional<float> maxValue;
+
+    // If a property is bound to this axis, this says whether the property changes should
+    // be forwarded to other connected nodes or session (similarly to \"isLocal\" for
+    // actions)
+    std::optional<bool> isRemote;
+};
+
+// @TODO: How to create a reference to the above object as the return type?
```

Comment:
@ylvaselling is this possible in the new documentation?  I feel like I need to discuss how to nicely handle return values that represent a certain type of objects